src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -155,20 +174,36 @@ public async Task<UpsertOutcome> UpsertAsync(ResourceWrapper resource, WeakETag
                             throw new MethodNotAllowedException(Core.Resources.ResourceCreationNotAllowed);
                         case SqlErrorCodes.TimeoutExpired:
                             throw new RequestTimeoutException(Resources.ExecutionTimeoutExpired);
+                        case 50400: // TODO: Add this to SQL error codes in AB#88286
```

Comment:
The PR with this change is [here](https://github.com/microsoft/healthcare-shared-components/pull/400).